src/tooling.git.md

```diff
@@ -0,0 +1,27 @@
+### Git branching model `[tooling.git]`
+
+We strive to follow [trunk-based development](https://trunkbaseddevelopment.com/) with few caveats.
+Since the cost of introducing correctness (or security) issues in our software is very high, we don't welcome even temporary quality regressions or sources of risk within our main branch. This may lead to feature branches that are a bit longer-lived than the trunk-based development ideal of 2 to 3 days.
+
+Generally, all significant code changes are reviewed by at least one team member and must pass CI.
+
+* For style and other trivial fixes, no review is needed (passing CI is sufficient).
+* For small ideas, use a PR.
+* For big ideas, create a RFC issue describing the goals before too much time is committed to the implementation.
+
+Feature branches are usually squashed and rebased upon merge to facilitate easier revering in case of discovered problems.
+
+Work on big ideas is divided into smaller, at most week-sized, chunks that each reasonably can be digested and reviewed in one sitting - the output here might be an overview, a prototype or progress towards a milestone or a snapshot of the work done so far, with the aim to involve reviewers early and allow space and time for course adjustments as the idea develops.
+
+Uncontested portions of the developed code that don't introduce significant risks or can be properly gated through feature flags are merged as soon as possible in order to reduce the size of the long-lived feature branch. Compile-time feature flags are preferred to run-time feature flags.
```

Comment:
this last bit is not necessarily true - a good example has been the introduction of the new connection model - it took user feedback to come up with adjustments that will need to be done, and such feedback cannot be expected to happen with compile-time flags. In general, now that the software is released, a practice we've been introducing is releasing certain features as preview behind a "beta" tag - this has worked pretty well.